### ISSUE TITLE
fileSlug should include respect folders with leading ISO dates

### ISSUE THREAD

**Issue by Brixy** [11ty/eleventy#1742]
Eleventy’s flexibility is great.

So far `fileSlug` only works for—well, it’s a telling name—file names, but should be extended to be used with folders.

This is a common and very logical way to structure posts and associated media:

```
.
└─ src/
   └─ posts/
      └─ 2021-04-20-a-post-without-media.md # not a folder
      └─ 2021-04-20-another-post/
      │  ├─ index.md
      │  ├─ image.jpg
      │  └─ audio.mp3
      └─ 2021-04-21-a-third-post/
         ├─ index.md
         └─ image.png
```

A simple `permalink` redirect is not possible in `posts.json` because of the mixture of `.md` files and folders with leading ISO dates. (Yet, Eleventy seems to recognize dates from folder names.)

It is easy to accomplish something like that using a filters for `permalink`.

But you run into problems due to #1140.

Thank you!

**Comment by Brixy** [comment]
I just happened to find out that eleventy **does** get the date from a folder name. Perfect!

The file `/posts/2021-07-01-my-post/index.md` has the following date (without adding `date` in front matter):

> (Thu Jul 01 2021 00:00:00 GMT+0200 (Mitteleuropäische Sommerzeit)

Should we clarify this?:

- Is it the expected behaviour that Eleventy gets the date from a folder name?
- Will this work in the future?

To get rid of the date from the path I just added a little filter:

```
eleventyConfig.addFilter('removeDateFromPath', (value) => {
  return value.replace(/\d{4}-\d{2}-\d{2}-/, "");
});
```
